File: spec/functional_test/fixtures/dart/cli_args/tool.dart
import 'package:args/args.dart';
import 'dart:io';

void main(List<String> args) {
  final parser = ArgParser();
  parser.addFlag('verbose', abbr: 'v');
  parser.addOption('name', abbr: 'n');

  final serve = parser.addCommand('serve');
  serve.addOption('port');

  final token = Platform.environment['API_TOKEN'];
  print(token);
}

File: spec/functional_test/fixtures/cpp/cli_cli11/tool.cpp
#include <CLI/CLI.hpp>
#include <cstdlib>
#include <string>

int main(int argc, char** argv) {
    CLI::App app{"My tool"};
    app.add_flag("-v,--verbose");

    std::string port;
    std::string config_file;

    auto* serve = app.add_subcommand("serve", "start the server");
    serve->add_option("-p,--port", port);
    serve->add_option("config", config_file);

    const char* token = std::getenv("API_TOKEN");
    CLI11_PARSE(app, argc, argv);
    return 0;
}
